app/oauth.go

```diff
@@ -202,26 +258,26 @@ func (a *App) GetOAuthAccessTokenForCodeFlow(clientId, grantType, redirectUri, c
 		return nil, model.NewAppError("GetOAuthAccessToken", "api.oauth.get_access_token.disabled.app_error", nil, "", http.StatusNotImplemented)
 	}
 
-	oauthApp, err := a.Srv().Store.OAuth().GetApp(clientId)
-	if err != nil {
+	oauthApp, nErr := a.Srv().Store.OAuth().GetApp(clientId)
+	if nErr != nil {
 		return nil, model.NewAppError("GetOAuthAccessToken", "api.oauth.get_access_token.credentials.app_error", nil, "", http.StatusNotFound)
 	}
 
 	if oauthApp.ClientSecret != secret {
 		return nil, model.NewAppError("GetOAuthAccessToken", "api.oauth.get_access_token.credentials.app_error", nil, "", http.StatusForbidden)
 	}
 
-	var user *model.User
 	var accessData *model.AccessData
 	var accessRsp *model.AccessResponse
 	if grantType == model.ACCESS_TOKEN_GRANT_TYPE {
 		var authData *model.AuthData
-		authData, err = a.Srv().Store.OAuth().GetAuthData(code)
-		if err != nil {
+		authData, nErr = a.Srv().Store.OAuth().GetAuthData(code)
+		if nErr != nil {
 			return nil, model.NewAppError("GetOAuthAccessToken", "api.oauth.get_access_token.expired_code.app_error", nil, "", http.StatusBadRequest)
 		}
 
 		if authData.IsExpired() {
+			// review: missing error handling, is it worth adding at least a warn log ?
```

Comment:
This is expected behavior for the regular usage of the application, I would add a `log.Debug`, or maybe (I'm not sure) a `mlog.Info`, but I don't think this should be a `Warn` log, because this is nothing wrong with this, there is nothing to wrong happening here. Let's see what @agnivade thinks about it.

Reply:
Hmm, any reason "there is nothing wrong happening here"? The SQL query can fail for any reason, and we should log that. IMO, a warn log should be there.

Reply:
mmm ok just to keep moving forward I will add an warn log (I agree with Agniva) but if we are still to disagree we can discuss it in the next review